### ISSUE TITLE
UR5's links and collision obstacles not aligned

### ISSUE THREAD

**Issue by LuciAnge** [uwgraphics/relaxed_ik#29]
Hi,
I am running the start_here.py in a ROS Melodic enviroment using Python2.7 to generate all necessary files to use [relaxed_ik_core.](https://github.com/uwgraphics/relaxed_ik_core) I am using a modified UR5 urdf ([my_ur5.txt](https://github.com/uwgraphics/relaxed_ik/files/7475487/my_ur5.txt)) because I need a specific position of end-effector link. Anyway, also with your ur5.urdf  the problem that I will present below was the same.
All seems fine until step 7. When I display robot platform with collision objects they don't match at all with robot's link. Seen below in the screen shot. **How can I fix this?**

![issue-photo](https://user-images.githubusercontent.com/90377166/140330905-7fc1b6c3-f1b4-4493-a48d-fcdc7274037b.PNG)

I assume that when the algorithm calculates distance between links, it actually calculates distance between their approximations (green cylindres). So if they don't match with robot's links all the process is useless. **Is it correct?**
Thank you for your time in advance.

**Comment by djrakita** [comment]
You are right, it is the green cylinders that trains the collision model.  Could you please send the urdf you are using, I'll try to figure out what's causing the mis-alignment.  Thanks

**Comment by djrakita** [comment]
You are right, it is the green cylinders that trains the collision model.  I'll try to figure out what's causing the mis-alignment using the urdf you included. Thanks

**Comment by LuciAnge** [comment]
Thank you for your reply.
I also tried to find a solution for the mis-alignment. Reading your code I figured out the way cylinders are generated. Every cylinder is defined thanks to two points superimposed on joint frames.

![issue-photo5](https://user-images.githubusercontent.com/90377166/141675088-225b599c-305e-491c-85dd-eec46ccea810.PNG)

I fixed the mis-alignment editing your code. My solution works only for UR5 and I'm sure it's not the best one. If you have any type of suggestion please let me know. These are the files that I modified [collision_utils.py](https://github.com/uwgraphics/relaxed_ik/files/7533468/collision_utils.txt) and [arm.py](https://github.com/uwgraphics/relaxed_ik/files/7533469/arm.txt).

![issue-photo4](https://user-images.githubusercontent.com/90377166/141675326-f111e8f0-7801-49b4-82bb-75abc5d22c19.PNG)

Thank you really much for your time.
